src/main/java/org/jabref/logic/xmp/DublinCoreExtractor.java

```diff
@@ -81,20 +93,41 @@ private void extractAuthor() {
     private void extractYearAndMonth() {
         List<String> dates = dcSchema.getUnqualifiedSequenceValueList("date");
         if ((dates != null) && !dates.isEmpty()) {
-            String date = dates.get(0).trim();
+
+            String dateString = dates.get(0).trim();
+            var dateFormat = ((DublinCoreSchemaCustom) dcSchema).getDateAttributeFormatQualifier();
```

Comment:
I still don't understand why it's necessary to add an additional attribute and modify the existing schema. This will probably break things.
Have you looked at JabRefs' Date.parse() it can already treat date ranges

Reply:
Hm, what does Acrobat show for the date value? Is there really still a difference in the month counting?

The biblatex Date Field supports Year month and date. And the field is also defined as an alias field for getting the month. getFieldOrAliasLatexFree 

An example from the office importer:
```
   result.day = entry.getFieldOrAliasLatexFree(StandardField.DAY).orElse(null);
        result.month = entry.getMonth().map(Month::getFullName).orElse(null);

        if (!entry.getLatexFreeField(StandardField.YEAR).isPresent()) {
            result.year = entry.getFieldOrAliasLatexFree(StandardField.YEAR).orElse(null);
        }
```

Reply:
Try to add a pattern like `uuuu-MM-dd'T'HH:mm:ssxxx` not sure about the xx

Reply:
I am currently testing this if this could work

Reply:
That's not the issue really @Siedlerchr. While we can parse the Date, the problem is that when we have the dc:date value as "2015-02" or "2015", the getElementsAsString returns defaults for the month and/or day values. That can be seen in the example screenshot. 
Eg : For Date : new Date(Year :2005), we would get 2005-01-01T00:00:00+05:30 from getElementsAsString

Reply:
Ah yes now I understand the issue but I also looked at the raw XMP xml string in getXmpMetadata  which is printed out. Maybe you could to a kind of heuristic: If  both month + date resolve to January 1st, that is unlikely, I would skip that field.
This is the raw output of the xml before any interpretation is happening.
```
 <dc:date>
        <rdf:Seq>
          <rdf:li>2018-03-01T00:00:00+01:00</rdf:li>
        </rdf:Seq>
      </dc:date>
```

Reply:
Well  yeah if it's Jan 1st, is likely I guess to be a default value set my the date converter, but then for the case where month and year are set, we should just to accept the value as is.
Anyways, I could go with that , but before that's done, I'm able to get the raw value by using a custom dcSchema that extends dcSchema and overrides the function responsible for getting the date value. Would like to know if that approach is fine or not

Reply:
>Well yeah if it's Jan 1st, It's likely, I guess, to be a default value set by the date converter, but then for the case where month and year are set, should we accept the date string as is?

Yes that sounds like a good way.

The date can be in different variations (The ISO8601 defines the standard date pattern and also the extended and partial variants as well)  Most common patterns are handled by JabRef's Date-Parser (see my comment above)
https://www.dublincore.org/specifications/dublin-core/dcmi-terms/terms/date/ 

So this is totally correct also for writing. You only write the information you have.

Reply:
Well, I get the heauristic approach part but I would like to know about whether the creation of the custom DcSchema class is fine or not. It uses FieldUtils because there aren't any setters for those fields.
I use this to get the raw value of the xmp date:dc li field, and not the getElementsAsString version ( which is, as you said,  ISO8601 converted,  henece the defaults). While getElementsAsStrijg provides us the iso8601 date time string, the raw value is as is from the xmp data. Could be iso8601, yyyy-mm, yyyy, yyyy-mm-dd, etc. Since we're handling parsing anyway, we might as well disallow the standard parsing done by dcSchema.

Reply:
Ah yes, I just checked that as well. So your approach looks good with the custom schema

Reply:
> Ah yes, I just checked that as well. So your approach looks good with the custom schema

Okay then, if everything else is good to go, I'll begin squashing commits, adding the "Support Date conversion from XMP to bibTex" in changelog. I don't think I need to add more tests honestly, since modifications to current Test Cases should cover that. There are no UI changes present. I guess the only thing remaining would be to run the Jabref application

Reply:
You could add one test for the date thing now. Maybe in the Writer Tests 
You also don't need to squash your commits. We do that usually on merge

Reply:
Oh...I already squashed them...
Anyways here's a screenshot of it working
![image](https://user-images.githubusercontent.com/4951690/156226092-6e36c6f1-13a4-4b1e-bb68-9af0d350cffe.png)

Sure, I'll add another test for that